security/pkg/nodeagent/plugin/providers/google/stsclient/stsclient.go

```diff
@@ -57,67 +53,86 @@ type federatedTokenResponse struct {
 
 // TokenExchanger for google securetoken api interaction.
 type Plugin struct {
-	hTTPClient *http.Client
+	httpClient  *http.Client
+	credFetcher security.CredFetcher
+	backoff     time.Duration
 }
 
 // NewPlugin returns an instance of secure token service client plugin
-func NewPlugin() security.TokenExchanger {
-	caCertPool, err := x509.SystemCertPool()
-	if err != nil {
-		stsClientLog.Errorf("Failed to get SystemCertPool: %v", err)
-		return nil
-	}
-	return Plugin{
-		hTTPClient: &http.Client{
-			Timeout: httpTimeOutInSec * time.Second,
-			Transport: &http.Transport{
-				TLSClientConfig: &tls.Config{
-					RootCAs: caCertPool,
```

Comment:
Forgot to ask, is it still using TLS or plain text now?

Reply:
Ah forgot I changed this; yes, the behavior is the same, the http default transport will use TLS whenever `https://` scheme is present and default to system root ca pool (like we manually set up here).

So should be identical behavior